dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Ticker;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Delegates execution to the given {@link LimitedChannel} until a failure is seen, at which point the channel will
+ * return {@link Optional#empty()} until the given {@link Duration} has elapsed.
+ */
+final class BlacklistingChannel implements LimitedChannel {
+
+    private static final String KEY = "key";
+    private static final String VALUE = "value";
+
+    private final LimitedChannel delegate;
+    private final Cache<String, String> isBlacklisted;
+
+    BlacklistingChannel(LimitedChannel delegate, Duration duration) {
+        this(delegate, duration, Ticker.systemTicker());
+    }
+
+    @VisibleForTesting
+    BlacklistingChannel(LimitedChannel delegate, Duration duration, Ticker ticker) {
+        this.delegate = delegate;
+        this.isBlacklisted = Caffeine.newBuilder()
+                .maximumSize(1)
+                .expireAfterWrite(duration)
+                .ticker(ticker)
+                .build();
+    }
+
+    @Override
+    public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
+        if (isBlacklisted.getIfPresent(KEY) != null) {
+            return Optional.empty();
+        } else {
+
+            return delegate.maybeExecute(endpoint, request)
+                    .map(future -> MoreFutures.addDirectCallback(future, new BlacklistingCallback<>()));
+        }
+    }
+
+    private class BlacklistingCallback<T> implements FutureCallback<T> {
+        @Override
+        public void onSuccess(T result) {}
+
+        @Override
+        public void onFailure(Throwable throwable) {
```

Comment:
does onFailure actually cover all the scenarios we care about?

Reply:
... this may lead back to the discussion about the `Call<T>` interface that gives you more fine-grained control over the type of error you're seeing

Reply:
If there are certain responses we'd like to blacklist for (eg: on 503s) we can do that in the onSuccess handler. @uschi2000 my claim is that we can do all the dispatching in our channels based on response code (and maybe will need headers) but don't need anything conjure specific. Basically, I'd like to keep our http channel stack completely conjure agnostic, given nothing about it should be conjure specific

Reply:
that's note quite true. for instance, this library might do something with 308, but not with 307, simply because conjure only cares about the former.

Reply:
I guess I'm fine saying that this library implements a subset of the http spec that is sufficient for the conjure use case. I mainly don't think it's worth taking a dependency on any sort of serialization logic in this library.